src/DataFormatter/DataFormatter.php

```diff
@@ -0,0 +1,330 @@
+<?php
+
+namespace SilverStripe\RestfulServer\DataFormatter;
```

Comment:
Later:

While we have the chance to do so, I think this class should be in the root package's namespace, then implementations of DataFormatter can live in that namespace. E.g.:

* SilverStripe\RestfulServer\DataFormatter
* SilverStripe\RestfulServer\DataFormatter\XMLDataFormatter
* SilverStripe\RestfulServer\DataFormatter\JSONDataFormatter